syntax = "proto3";

package fileupload;

service FileUploadService {
  rpc UploadFile(stream UploadRequest) returns (UploadResponse);
  rpc GetFileURL(FileRequest) returns (FileURLResponse);
  rpc MergeChunks(FileRequest) returns (UploadResponse);
}

message UploadRequest {
  string fileName = 1;
  int32 chunkIndex = 2;
  bytes content = 3;
}

message UploadResponse {
  string message = 1;
}

message FileRequest {
  string fileName = 1;
}

message FileURLResponse {
  string fileUrl = 1;
}
